### ISSUE TITLE
[Bug]: Transform runner does not work if one of the inputs is a Object Storage V2 materialization

### ISSUE THREAD

**Issue by frapa** [emdgroup/foundry-dev-tools#26]
### Issue checklist

- [X] This is a **bug** in Foundry DevTools and not a bug in another project. It is also not an enhancement/feature request
- [X] I searched through the GitHub issues and this issue has not been opened before.
- [X] I have installed the latest version of Foundry DevTools and don't use an unsupported python version.

### Description of the bug

I tried running transforms as described in the docs, and the package works wonderfully. However, the package fails if one of the transform inputs is a materialization of a Object Storage V2 table (writeback). I guess the reason is that in V2, materializations aren't datasets anymore but views. As such, they probably require another endpoint to read the data from.

### Steps to reproduce this bug.

I have a transform such as:

```python
from pyspark.sql import DataFrame
from transforms.api import transform_df, Input, Output

@transform_df(
    Output("..."),
    input_df=Input("<RID-of-V2-materialization>"),
)
def compute(input_df: DataFrame) -> DataFrame:
    return input_df

if __name__ == "__main__":
    compute.compute().show()
```

Running the script it fails with the error below.

### Log output

```
<removed>\venv\Scripts\python.exe C:/srdev/projects/am_projection/transforms-python/src/am_projection/transforms/quarterly_projection.py
c:\srdev\projects\foundrycommons\src\stargatecommons\df_utils\dataset_metadata.py:12: UserWarning: Cannot import foundry transform package locally, skipping. Some functionality might not work as expected.
  warnings.warn(
<removed>\venv\lib\site-packages\pyspark\sql\pandas\utils.py:24: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
500 Server Error: Internal Server Error for url: https://stargate.swissre.com/foundry-data-proxy/api/dataproxy/datasets/ri.foundry.main.dataset.d129cc10-11a3-4cb6-9978-4ff56482f5f8/views/master/edits_empty_file_edb25a13-5503-438b-8bfc-635ce144c929
{"errorCode":"INTERNAL","errorName":"Default:Internal","errorInstanceId":"44e9cce6-146c-40bc-810a-e9fc9b938158","parameters":{}}
Traceback (most recent call last):
  File "<removed>\transforms-python\src\am_projection\transforms\quarterly_projection.py", line 13, in <module>
    assumptions=Input("ri.foundry.main.dataset.d129cc10-11a3-4cb6-9978-4ff56482f5f8"),
  File "<removed>\venv\lib\site-packages\transforms\api\_dataset.py", line 93, in __init__
    self._spark_df, self._dataset_identity, self.branch = self._online(
  File "<removed>\venv\lib\site-packages\transforms\api\_dataset.py", line 123, in _online
    self._retrieve_spark_df(dataset_identity, branch),
  File "<removed>\venv\lib\site-packages\transforms\api\_dataset.py", line 160, in _retrieve_spark_df
    return self._retrieve_from_foundry_and_cache(dataset_identity, branch)
  File "<removed>\venv\lib\site-packages\transforms\api\_dataset.py", line 205, in _retrieve_from_foundry_and_cache
    spark_df = self._cached_client.load_dataset(
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\cached_foundry_client.py", line 80, in load_dataset
    _, dataset_identity = self.fetch_dataset(dataset_path_or_rid, branch)
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\cached_foundry_client.py", line 116, in fetch_dataset
    self._download_dataset_and_return_local_path(
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\cached_foundry_client.py", line 148, in _download_dataset_and_return_local_path
    self._download_dataset_to_cache_dir(dataset_identity, branch, foundry_schema)
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\cached_foundry_client.py", line 175, in _download_dataset_to_cache_dir
    self.api.download_dataset_files(**params)
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\foundry_api_client.py", line 1195, in download_dataset_files
    self.download_dataset_file(
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\foundry_api_client.py", line 1136, in download_dataset_file
    resp = self._download_dataset_file(
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\foundry_api_client.py", line 1155, in _download_dataset_file
    _raise_for_status_verbose(response)
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\foundry_api_client.py", line 2067, in _raise_for_status_verbose
    raise error
  File "<removed>\venv\lib\site-packages\foundry_dev_tools\foundry_api_client.py", line 2056, in _raise_for_status_verbose
    response.raise_for_status()
  File "<removed>\venv\lib\site-packages\requests\models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://stargate.swissre.com/foundry-data-proxy/api/dataproxy/datasets/ri.foundry.main.dataset.d129cc10-11a3-4cb6-9978-4ff56482f5f8/views/master/edits_empty_file_edb25a13-5503-438b-8bfc-635ce144c929
```

### Additional context

pyspark 3.2.0

### Operating System

Windows

### Your python version

3.10.4

**Comment by jonas-w** [comment]
Thank you for raising this issue. We are aware of this limitation with views and are already working on lifting this limitation.

**Comment by nicornk** [comment]
@frapa Can you give this branch a try and report feedback?
https://github.com/emdgroup/foundry-dev-tools/tree/feat/view-support

After cloning you can install into your environment with `pip install -e .` when in the foundry-dev-tools directory.

One issue is that we don't know the file size of the view while querying, so it will potentially pull down large datasets - which might make it look like the process is stuck.

**Comment by nicornk** [comment]
closed by #27 and now available in v1.2 https://pypi.org/project/foundry-dev-tools/1.2/
